site_scons/site_tools/intel_c.py

```diff
@@ -0,0 +1,657 @@
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+from __future__ import division, print_function
+
+__revision__ = "src/engine/SCons/Tool/intelc.py  2020-01-22 06:48:41 mockbuild"
+
+import math, sys, os.path, glob, string, re
+
+is_windows = sys.platform == 'win32'
+is_win64 = is_windows and (os.environ['PROCESSOR_ARCHITECTURE'] == 'AMD64' or
+                           ('PROCESSOR_ARCHITEW6432' in os.environ and
+                            os.environ['PROCESSOR_ARCHITEW6432'] == 'AMD64'))
+is_linux = sys.platform.startswith('linux')
+is_mac = sys.platform == 'darwin'
+
+if is_windows:
+    import SCons.Tool.msvc
+elif is_linux:
+    import SCons.Tool.gcc
+elif is_mac:
+    import SCons.Tool.gcc
+import SCons.Util
+import SCons.Warnings
+
+
+# Exceptions for this tool
+class IntelCError(SCons.Errors.InternalError):
+    pass
+
+
+class MissingRegistryError(IntelCError):  # missing registry entry
+    pass
+
+
+class MissingDirError(IntelCError):  # dir not found
+    pass
+
+
+class NoRegistryModuleError(IntelCError):  # can't read registry at all
+    pass
+
+
+def linux_ver_normalize(vstr):
+    """Normalize a Linux compiler version number.
+    Intel changed from "80" to "9.0" in 2005, so we assume if the number
+    is greater than 60 it's an old-style number and otherwise new-style.
+    Always returns an old-style float like 80 or 90 for compatibility with Windows.
+    Shades of Y2K!"""
+    # Check for version number like 9.1.026: return 91.026
+    # XXX needs to be updated for 2011+ versions (like 2011.11.344 which is compiler v12.1.5)
+    m = re.match(r'([0-9]+)\.([0-9]+)\.([0-9]+)', vstr)
+    if m:
+        vmaj, vmin, build = m.groups()
+        return float(vmaj) * 10. + float(vmin) + float(build) / 1000.
+    else:
+        f = float(vstr)
+        if is_windows:
+            return f
+        else:
+            if f < 60:
+                return f * 10.0
+            else:
+                return f
+
+
+def check_abi(abi):
+    """Check for valid ABI (application binary interface) name,
+    and map into canonical one"""
+    if not abi:
+        return None
+    abi = abi.lower()
+    # valid_abis maps input name to canonical name
+    if is_windows:
+        valid_abis = {'ia32': 'ia32',
+                      'x86': 'ia32',
+                      'ia64': 'ia64',
+                      'em64t': 'em64t',
+                      'amd64': 'em64t'}
+    if is_linux:
+        valid_abis = {'ia32': 'ia32',
+                      'x86': 'ia32',
+                      'x86_64': 'x86_64',
+                      'em64t': 'x86_64',
+                      'amd64': 'x86_64'}
+    if is_mac:
+        valid_abis = {'ia32': 'ia32',
+                      'x86': 'ia32',
+                      'x86_64': 'x86_64',
+                      'em64t': 'x86_64'}
+    try:
+        abi = valid_abis[abi]
+    except KeyError:
+        raise SCons.Errors.UserError("Intel compiler: Invalid ABI %s, valid values are %s" % \
```

Comment:
(pylint-line-too-long) Line too long (94/80)
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword
(pylint-no-member) Module 'SCons.Errors' has no 'UserError' member